src/middlewared/middlewared/plugins/idmap.py

```diff
@@ -646,9 +687,23 @@ async def do_update(self, id, data):
                         'generate LDAP traffic. Certificates do not apply.')
         verrors.check()
         await self.prune_keys(new)
-        final_options = IdmapBackend[data['idmap_backend']].defaults()
+        final_options = IdmapBackend[new['idmap_backend']].defaults()
         final_options.update(new['options'])
         new['options'] = final_options
+
+        if new['options'].get('ldap_user_dn_password'):
+            try:
+                DSType[new["name"]]
+                domain = (await self.middleware.call("smb.config"))['workgroup']
+            except KeyError:
+                domain = new["name"]
+
+            secret = data['options'].pop('ldap_user_dn_password')
```

Comment:
Do we really want this to pop from `data`
And we are redefining secret on next line which I imagine is the only real thing we need